catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/testsuites/generateAllEspressoTestsSuite.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+#put header
+echo "/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */\n" > AllEspressoTestsSuite.java
+
+#put packagename
+echo "package org.catrobat.catroid.uiespresso.testsuites;\n" >> AllEspressoTestsSuite.java
+
+#put imports
+find .. -name '*Test.java' | sed 's/\.\.\//import org.catrobat.catroid.uiespresso./' | sed 's/\//\./g' | sed 's/.java/;/' | sort >> AllEspressoTestsSuite.java
+
+#put annotations
+echo "import org.junit.runner.RunWith;\nimport org.junit.runners.Suite;\n\n@RunWith(Suite.class)\n@Suite.SuiteClasses({" >> AllEspressoTestsSuite.java
+
+#put classes to be imported
+find .. -name '*Test.java' | sed 's/^.*\///' | sed 's/java/class/' | sed 's/$/,/' | sed '$s/,//' | sed 's/^/		/' >> AllEspressoTestsSuite.java
+
+#put class
+echo "})\npublic class AllEspressoTestsSuite {\n}" >> AllEspressoTestsSuite.java
+
+exit
```

Comment:
We could exclude this file entirely from Checkstyle [catroid/gradle/code_quality_tasks.gradle#L28](https://github.com/Catrobat/Catroid/blob/be9b08440d257bdb9adbd6cf142aae5b41a8e8d0/catroid/gradle/code_quality_tasks.gradle#L28).
However, if you reformat the whole project through AS, the import order will always be changed.

We should also add a comment to the java file that it's auto generated and that you should use the script to create it (and then run the formatter, if we don't go with excluding the file).

I think we should add a `| sort` [here](https://github.com/Catrobat/Catroid/blob/be9b08440d257bdb9adbd6cf142aae5b41a8e8d0/catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/testsuites/generateAllEspressoTestsSuite.sh#L4). Otherwise the order is completely different depending on the OS. Not sure why find is different on Ubuntu :anguished:

```diff
 @RunWith(Suite.class)
 @Suite.SuiteClasses({
-        PocketMusicActivityTest.class,
+        FormulaEditorKeyboardTest.class,
+        FormulaEditorTest.class,
         SmokeTest.class,
         FlakyTestTest.class,
-        WhenNfcBrickStageTest.class,
-        LegoEv3PlayToneBrickTest.class,
-        ComeToFrontBrickTest.class,
-        HideBrickTest.class,
-        ChangeTransparencyByNBrickTest.class,
+        PocketMusicActivityTest.class,
…
```

Since this task is executed quickly, maybe we add it to gradle to always run before any build?